### PR TITLE
Refactor interface for `instantaneous_zenith_angle` function

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Refactor interface for `instantaneous_zenith_angle` function
Tag v0.8.0

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/ZenithAngleCalc.jl [hunk 1]
```diff
@@ -51,7 +50,7 @@ function distance_declination_hourangle(
     d0 = IP.orbit_semimaj(param_set)
     M0 = IP.mean_anom_epoch(param_set)
 
-    date0 = DateTime(epoch_string, dateformat"y-m-dTHH:MM:SS.s")
+    #date0 = DateTime(epoch_string, dateformat"y-m-dTHH:MM:SS.s")
```
Comment (szy21) [inline src/ZenithAngleCalc.jl:53]:
```suggestion
```

**Review by szy21 — APPROVED** [review]

File: src/ZenithAngleCalc.jl [hunk 2]
```diff
@@ -30,19 +30,18 @@ compute_orbital_parameters(param_set) = (
 function get_Δt_years(
     param_set::IP.InsolationParameters{FT},
     date,
-    epoch_string = "2000-01-01T11:58:56.816",
+    date0,
```
Comment (simonbyrne) [inline src/ZenithAngleCalc.jl:33]:
```suggestion
    date_epoch::DateTime,
```

File: src/ZenithAngleCalc.jl [hunk 3]
```diff
@@ -86,7 +83,7 @@ end
 
 function instantaneous_zenith_angle(
     date::DateTime,
-    epoch_string,
+    date0::DateTime,
     (ϖ, γ, e)::Tuple{FT, FT, FT},
```
Comment (simonbyrne) [inline src/ZenithAngleCalc.jl:87]: what are these arguments? Are they the same at every lat/long point?
@szy21 do you know?

Comment (simonbyrne) [reply src/ZenithAngleCalc.jl:87]: (if they are the same, then we don't need to call `distance_declination_hourangle` at every point.)

Comment (szy21) [reply src/ZenithAngleCalc.jl:87]: ϖ, γ, e? They are orbital parameters (longitude of perihelion, obliquity, and eccentricity), and they are the same at every lat/long point. There is some documentation [here](https://clima.github.io/Insolation.jl/dev/ZenithAngleEquations/). I can add some doc string later, but probably not now.

**Comment by sriharshakandala** [comment]
@simonbyrne : Is this interface reasonable?
`d` can be omitted here. I currently left this in to stay consistent with the current return type! cc: @szy21

File: src/ZenithAngleCalc.jl [hunk 4]
```diff
@@ -84,27 +81,29 @@ function distance_declination_hourangle(
...
     latitude::FT,
     param_set::IP.AIP,
-    eot_correction::Bool,
 ) where {FT}
-    λ = deg2rad(longitude)
-    ϕ = deg2rad(latitude)
 
+Returns the zenith angle, azimuthal angle and earth-sun distance
+at a particular longitude and latitude on the given date (and time UTC)
+given distance, declination, hour angle.
```
Comment (szy21) [inline src/ZenithAngleCalc.jl:94]:
```suggestion
Returns the zenith angle, azimuthal angle and Earth-Sun distance
at a particular longitude and latitude given Earth-Sun distance, declination angle, 
and hour angle at longitude = 0.
```

**Comment by simonbyrne** [comment]
from what i can tell, it looks fine, but i would suggest making the corresponding changes to ClimaAtmos, and see if it runs

**Comment by sriharshakandala** [comment]
> from what i can tell, it looks fine, but i would suggest making the corresponding changes to ClimaAtmos, and see if it runs

Yes. I am already working on that!

**Comment by sriharshakandala** [comment]
I tried this locally on the cluster and it works! 
I made a draft PR at https://github.com/CliMA/ClimaAtmos.jl/pull/2341
I started the GPU pipeline here: https://buildkite.com/clima/climaatmos-target-gpu-simulations/builds/134

**Comment by sriharshakandala** [comment]
It runs to completion on the GPU pipeline here: https://buildkite.com/clima/climaatmos-target-gpu-simulations/builds/138#018bad4c-6dfa-42fa-89fc-48292ca85d8b

**Comment by sriharshakandala** [comment]
I am planning to merge unless there are any additional comments!

**Comment by simonbyrne** [comment]
Bump the version?

**Comment by sriharshakandala** [comment]
> Bump the version?

Done.

File: src/ZenithAngleCalc.jl [hunk 5]
```diff
@@ -84,27 +81,28 @@ function distance_declination_hourangle(
...
+    instantaneous_zenith_angle(
+        (d, δ, η_UTC)::Tuple{FT, FT, FT},
+        longitude::FT,
+        latitude::FT,
+        param_set::IP.AIP,
+    ) where {FT}
+
+Returns the zenith angle, azimuthal angle and Earth-Sun distance
+at a particular longitude and latitude given Earth-Sun distance, declination angle, 
+and hour angle at longitude = 0.
```
Comment (simonbyrne) [inline src/ZenithAngleCalc.jl:94]: Good to clarify units here...
```suggestion
Returns the zenith angle, azimuthal angle (both in radians) and Earth-Sun distance
at a particular longitude and latitude (in degrees) given Earth-Sun distance, declination angle (in radians), 
and hour angle at longitude = 0.
```

File: src/ZenithAngleCalc.jl [hunk 6]
```diff
@@ -84,27 +81,28 @@ function distance_declination_hourangle(
     return d, δ, η_UTC
 end
 
+"""
+    instantaneous_zenith_angle(
+        (d, δ, η_UTC)::Tuple{FT, FT, FT},
```
Comment (simonbyrne) [inline src/ZenithAngleCalc.jl:86]: This doesn't match the function signature below

**Pull request merged**